File: tests/patterns/typescript/aws_dynamodb_to_firestore/after.ts
import { Firestore } from "@google-cloud/firestore";

const db = new Firestore();
const COLLECTION_NAME = "users";

interface User {
  userId: string;
  name: string;
  email: string;
}

export async function createUser(user: User): Promise<void> {
  const docRef = db.collection(COLLECTION_NAME).doc(user.userId);
  await docRef.set(user);
}

export async function getUser(userId: string): Promise<User | null> {
  const docRef = db.collection(COLLECTION_NAME).doc(userId);
  const doc = await docRef.get();
  if (!doc.exists) return null;
  return doc.data() as User;
}

export async function deleteUser(userId: string): Promise<void> {
  const docRef = db.collection(COLLECTION_NAME).doc(userId);
  await docRef.delete();
}

export async function queryUsersByEmail(email: string): Promise<User[]> {
  const snapshot = await db
    .collection(COLLECTION_NAME)
    .where("email", "==", email)
    .get();
  return snapshot.docs.map((doc) => doc.data() as User);
}

File: tests/patterns/python/aws_dynamodb_to_firestore/after.py
from google.cloud import firestore

db = firestore.Client()
collection = db.collection('users')


def create_user(user_id, name, email):
    doc_ref = collection.document(user_id)
    doc_ref.set({
        'name': name,
        'email': email,
    })


def get_user(user_id):
    doc_ref = collection.document(user_id)
    doc = doc_ref.get()
    if doc.exists:
        return doc.to_dict()
    return None


def update_user(user_id, name):
    doc_ref = collection.document(user_id)
    doc_ref.update({
        'name': name,
    })


def delete_user(user_id):
    doc_ref = collection.document(user_id)
    doc_ref.delete()


def query_users_by_email(email):
    query = collection.where('email', '==', email)
    docs = query.stream()
    return [doc.to_dict() for doc in docs]
